src/main/ngapp/accio/src/app/editor/editor.component.ts

```diff
@@ -15,148 +14,200 @@ import { ImageBlob } from '../ImageBlob';
 export class EditorComponent implements OnInit {
   constructor(
     private route: ActivatedRoute,
-    private router: Router,
     private postBlobsService: PostBlobsService,
   ) { }
 
-  url: string;
-  display: boolean = false;
-  projectId: string;
+  //  Display variables.
+  private image: HTMLImageElement;
+  imageUrl: string;
+  displayMaskForm: boolean = false;
+  disableSubmit: boolean = false;
+
+  //  Mask variables.
+  private maskImageData: ImageData;
+  private maskImageUrl: string
   
+  //  Form variables.
   uploadMaskForm: FormGroup;
   formData: FormData;
+  projectId: string;
   parentName: string;
-  blobMask;
+  blobMask: Blob; 
 
+  //  scaleFactor is used to trim image scale so image 
+  //    is smaller than width and height of the users screen.
+  //  The following variables are binded to their 
+  //    respective inputs in MaskDirective and change in mask.directive
+  //    when they change in editor.component
+  scaleFactor: number;
   originalImageData: ImageData;
-  scaledImageData: ImageData;
+  tolerance: number;
+  disableFloodFill: boolean;
+  maskTool: string;
```

Comment:
It would be helpful to have a comment of what this is.

It would also help if you used a string enum as shown here: https://www.typescriptlang.org/docs/handbook/enums.html
(But that can be a follow-up PR if you prefer.)

Reply:
Noted in the comments as a TODO. Will work on that this morning.